vtex-sites/base.store
# chore: Remove gatsby-plugin-offline

## What's the purpose of this pull request?
This PR removes the `gatsby-plugin-offline` plugin

## How does it work?
Gatsby plugin offline is the recommended way of adding service worker support to Gatsby. It caches important assets, like JavaScript and CSS for faster loading times on recurring users. Also, this plugin implements a pattern called [app-shell pattern](https://developers.google.com/web/fundamentals/architecture/app-shell) that aims to give users the impression of faster loading times by rendering the apps shell before the app's content. 
The way this plugin implements this pattern is by intercepting any reload the user does, returning the app shell and doing a client-side navigation to the desired page. This works well on some projects, but on ours, this pattern was creating a huge layout shift. The left side of the image below shows this LS, where the app's shell is loaded, then a client side navigation is performed and the app's content is loaded. The image on right shows the same device performing the load without this plugin.
<div style="display: flex; justify-content: center">
<img width="300px" src="https://user-images.githubusercontent.com/1753396/154575229-106e25b7-1562-4416-824a-317f5dc471e3.gif"/> 
<img width="300px" src="https://user-images.githubusercontent.com/1753396/154575222-b80951fc-5c06-4bdd-bb4d-44bfd9ee2b3a.gif"/>
</div>

Also, I noticed that another of solving this problem is by adding a `min-height` to the `<main/>` tag so that this CLS is not visible. 

I think PWA support is not currently necessary and we may go back to this subject once we have fixed other performance issues later on. Also, I noticed that Gatsby's lighthouse score increased after removing this plugin. I'd say we remove this for now and go back to it later on.

> Note that the service worker I'm removing don't really affect partytown or any other pixel scripts

## How to test it?
Make sure you have enabled service workers and cache on devTools. Also, go to the `performance` tab and throttle your CPU 6x. Perform a reload of this PR vs master. You will see the difference

## Checklist
